sync-core/src/main/java/com/cloudant/sync/query/UnindexedMatcher.java

```diff
@@ -304,23 +322,73 @@ protected static boolean compareNE(Object l, Object r) {
     //  3. TEXT
     //  4. BLOB
     protected static boolean compareLT(Object l, Object r) {
-        // TODO: 275 - 309 CDTQUnindexedMatcher.m
-        return false;
+        if (l == null) {
+            return false;  // null fails all lt/gt/lte/gte tests
+        } else if (!(l instanceof String || l instanceof Number && !(l instanceof Float))) {
+            String msg = String.format("Value in document not a Number or String: %s", l);
+            logger.log(Level.WARNING, msg);
+            return false;  // Not sure how to compare values that are not numbers or strings
+        } else if (l instanceof String) {
+            if (r instanceof Number) {
+                return false;  // INT < STRING
+            }
+
+            String lStr = (String) l;
+            String rStr = (String) r;
+
+            return lStr.compareTo(rStr) < 0;
+        } else if (r instanceof String) {
+            // At this point in the logic l can only be a number
+            return true;  // INT < STRING
+        } else {
+            // At this point in the logic both l and r can only be numbers
+            Number lNum = (Number) l;
+            Number rNum = (Number) r;
+
+            return lNum.longValue() < rNum.longValue();
```

Comment:
Oh man, I missed that completely.  I fixed it in compareEq but left it here.  I'll fix it and add more tests to the unindexed matcher tests to specifically test all compare methods.